dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/builder/TaskExecutionContextBuilder.java

```diff
@@ -126,6 +127,24 @@ public TaskExecutionContextBuilder buildProcessDefinitionRelatedInfo(ProcessDefi
         return this;
     }
 
+    /**
+     * build execPath related info
+     *
+     * @return TaskExecutionContextBuilder
+     */
+    public TaskExecutionContextBuilder buildExecPathRelatedInfo() {
```

Comment:
I think it should setExecutePath and appInfoPath in worker before task running.  cc @ruanwenjun

Reply:
@caishunfeng 
I think whether `execPath` or `appInfoPath` is context-related, and will never change, it's weird to set it in worker, so i put it forward in master.

Reply:
> I think it should setExecutePath and appInfoPath in worker before task running. cc @ruanwenjun
Yes, it should be move to worker, and should be set before yarn task running, I am not sure if other task will need this, 
you can simply put this before all task running.

@Radeity In fact, the master doesn't know execPath, it is defined in worker, different worker may have different path, master doesn't know the execpath

Reply:
> > I think it should setExecutePath and appInfoPath in worker before task running. cc @ruanwenjun
> 
> Yes, it should be moved to worker and should be set before yarn task running, I am not sure if other task will need this, you can simply put this before all task running.
> 
> @Radeity In fact, the master doesn't know execPath, it is defined in worker, different worker may have different path, master doesn't know the execpath

Oops! I have ignored that different workers may have different `DATA_BASEDIR`. I'll change this part of code, but, i'll still keep `appInfoPath`, is that okay?